src/composables/useGetMessages.ts

```diff
@@ -539,6 +551,66 @@ export function useGetMessagesProvider() {
 		}, 500)
 	}
 
+	/**
+	 * Try to start chat relay
+	 */
+	function tryChatRelay() {
+		if (chatRelaySupported && isChatEndReached.value) {
+			startChatRelay()
+		}
+	}
+
+	/**
+	 * Check whether chat relay is supported
+	 *
+	 * @param features
+	 */
+	function checkChatRelaySupport(features: string[]) {
+		if (features.includes('chat-relay')) {
+			chatRelaySupported = true
+		}
+	}
+
+	/**
+	 * Initialize chat relay support by stopping polling and listening to chat relay messages
+	 */
+	function startChatRelay() {
+		if (currentToken.value) {
+			// it might have been set already, ensure we cancel it
+			store.dispatch('cancelPollNewMessages', { requestId: currentToken.value })
+		}
+		chatRelayEnabled = true
+		EventBus.on('chat-relay-message-received', addMessageFromChatRelay)
+	}
+
+	/**
+	 *
+	 * Chat relay sends one message at a time, we update our stores directly
+	 *
+	 * @param payload
+	 * @param payload.token
+	 * @param payload.message
+	 */
+	function addMessageFromChatRelay(payload: { token: string, message: ChatMessage }) {
+		const { token, message } = payload
+		if (token !== currentToken.value) {
+			// Guard: Message is for another conversation
+			// e.g., user switched conversation while messages were in-flight
+			return
+		}
+
+		chatStore.processChatBlocks(token, [message], { mergeBy: conversationLastMessageId.value })
```

Comment:
This is more correct, as computed will return you last thread message, if you're in the thread
```suggestion
		chatStore.processChatBlocks(token, [message], { mergeBy: chatStore.getLastKnownId(token) })
```